api/v1beta1/glance_types.go

```diff
@@ -72,6 +73,11 @@ type GlanceSpecCore struct {
 	// NodeSelector to target subset of worker nodes running this service
 	NodeSelector *map[string]string `json:"nodeSelector,omitempty"`
 
+	// +kubebuilder:validation:Optional
+	// TopologyRef to apply the Topology defined by the associated CR referenced
+	// by name
+	TopologyRef *topology.TopoRef `json:"topologyRef,omitempty"`
```

Comment:
I wondered why we need a TopologyRef in the GlanceCoreSpec structure when there's also one in the GlanceAPITemplate structure in common_types.go. That had me looking at the NodeSelector field (asking myself the same question; why is it needed in this structure when there's one in the GlanceAPITemplate?).

From what I can tell, the NodeSelector field in this structure is only referenced by the pkg code that handles the dbsync and cronjob stuff. I guess dbsync and cronjobs can't (re)use the NodeSelector from a GlanceAPI? That's a separate question unrelated to this PR, but it has me asking whether we need a TopologyRef in this structure. Will it be referenced anywhere, for example in some future update to the dbsync or cronjob code?